lib/framework/action.dart

```diff
@@ -359,6 +359,49 @@ class FileUploadAction extends EnsembleAction {
   }
 }
 
+class WalletConnectAction extends EnsembleAction {
+  WalletConnectAction({
+    this.id,
+    required this.wcProjectId,
+    required this.appName,
+    this.appDescription,
+    this.appUrl,
+    this.appIconUrl,
+    this.onComplete,
+    this.onError,
+  });
+
+  String? id;
+  String wcProjectId;
+  String appName;
+  String? appDescription;
+  String? appUrl;
+  String? appIconUrl;
+  EnsembleAction? onComplete;
+  EnsembleAction? onError;
+
+
+  factory WalletConnectAction.fromYaml({YamlMap? payload}) {
+    if (payload == null ||
+        (payload['wcProjectId'] == null && payload['name'] == null)) {
+      throw LanguageError(
+          "${ActionType.connectWallet.name} requires either a wcProjectId, appName, appDescription, appUrl, appIconUrl.");
+    }
+    return WalletConnectAction(
+      id: Utils.optionalString(payload['id']),
+      wcProjectId: payload['wcProjectId'],
```

Comment:
you'd want to use Utils.getString/optionalString since user can hose the system just by entering a number